packages/dds/tree/src/test/shared-tree/sharedTreeFuzzTests.ts

```diff
@@ -0,0 +1,371 @@
+            }
+            break;
+        default:
+            fail("Invalid edit.");
+    }
+}
+
+async function applyFuzzChange(tree: ISharedTree, contents: FuzzChange): Promise<void> {
+    const index = contents.index;
+    const nodeField = contents.field;
+    switch (contents.fuzzType) {
+        case "insert":
+            if (index !== undefined && nodeField !== undefined) {
+                tree.runTransaction((forest, editor) => {
+                    const field = editor.sequenceField(contents.path, nodeField);
+                    field.insert(
+                        index,
+                        singleTextCursor({ type: brand("Test"), value: contents.value }),
+                    );
+                    return TransactionResult.Apply;
+                });
+            }
+            break;
+        case "delete":
+            if (index !== undefined && nodeField !== undefined) {
+                const parent = contents.path?.parent;
+                const delField = contents.path?.parentField;
+                const parentIndex = contents.path?.parentIndex;
+                if (delField !== undefined && parent !== undefined && parentIndex !== undefined) {
+                    tree.runTransaction((forest, editor) => {
+                        const field = editor.sequenceField(parent, delField);
+                        field.delete(parentIndex, 1);
+                        return TransactionResult.Apply;
+                    });
+                }
+            }
+            break;
+        case "setPayload":
+            if (index !== undefined && nodeField !== undefined) {
+                const path = contents.path;
+                if (path !== undefined) {
+                    tree.runTransaction((forest, editor) => {
+                        editor.setValue(path, contents.value);
+                        return TransactionResult.Apply;
+                    });
+                }
+            }
+            break;
+        default:
+            fail("Invalid edit.");
+    }
+}
+
+export function runSharedTreeFuzzTests(title: string): void {
+    describeFuzz(title, ({ testCount }) => {
+        function runTest(
+            generatorFactory: () => AsyncGenerator<Operation, FuzzTestState>,
+            seed: number,
+        ): void {
+            for (let i = 0; i < 10; i++) {
+                it(`with seed ${i}`, async () => {
+                    await performFuzzActions(generatorFactory(), i);
+                }).timeout(20000);
+            }
+        }
+        function runTestAbort(
+            generatorFactory: () => AsyncGenerator<Operation, FuzzTestState>,
+            seed: number,
+        ): void {
+            for (let i = 0; i < 10; i++) {
+                it.skip(`with seed ${i}`, async () => {
+                    await performFuzzActionsAbort(generatorFactory(), i);
+                }).timeout(20000);
+            }
+        }
+        describe("with no-history summarization", () => {
```

Comment:
I'm guessing this title was copied over from legacy shared tree.
I would just name it "basic convergence".